model/userdata.go

```diff
@@ -147,3 +148,105 @@ func validateCoreosCloudInit(content []byte) error {
 func validateNone([]byte) error {
 	return nil
 }
+
+func (self UserDataPart) Base64UserDataRef() (string, error) {
+	// Change this to Base64IgnitionConfigRef to test with ignition
+	return self.Base64BashScriptRef()
+}
+
+func (self UserDataPart) Base64IgnitionConfigRef() (string, error) {
+	base64Part, err := self.Base64(false)
+	if err != nil {
+		return "", fmt.Errorf("failed to generate ignition config: %v", err)
+	}
+
+	// TODO Append user-provided ignition config via `ignition.config.append.source`
+	// once it supports sourcing from a s3 object
+	// https://coreos.com/ignition/docs/0.14.0/configuration-v2_0.html
+
+	head, err := json.Marshal(`{
+  "ignition": {
+    "version": "2.0.0"
+  },
+  "storage": {
+    "files": [
+      {
+        "filesystem": "root",
+        "path": "/kube-aws-stack-name",
```

Comment:
The path should be somewhere else like `/var/run/coreos/kube-aws-stack-name`. However, `/var/run` is a tmpfs other than the root filesystem, which requires me to provide ignition an another filesystem name other than `"root"`.
The problem doing so is that I'm not sure that names of filesystems other than `"root"` is stable i.e. safe to be hard-coded.
So, I'm rather wanting to leave this undecided for now.